### PR TITLE
Add polish to coroutine support in RESTEasy Reactive

### PR DESCRIPTION
**Pull request by geoand**

### CODE REVIEW

File: extensions/resteasy-reactive/quarkus-resteasy-reactive/deployment/src/test/java/io/quarkus/resteasy/reactive/server/test/kotlin/BlockingCoroutinesTest.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    public void test() {
+        fail("Should never be called");
+    }
+
+    @Path("test")
+    public static class TestEndpoint {
+
+        @Path("hello")
+        @Blocking
+        public Object hello(kotlin.coroutines.Continuation<?> cont) {
```
Comment (cescoffier) [inline extensions/resteasy-reactive/quarkus-resteasy-reactive/deployment/src/test/java/io/quarkus/resteasy/reactive/server/test/kotlin/BlockingCoroutinesTest.java:39]: I would rather avoid having Kotlin dependency in this module.

File: extensions/resteasy-reactive/quarkus-resteasy-reactive/kotlin/src/main/kotlin/org/jboss/resteasy/reactive/server/runtime/kotlin/ApplicationCoroutineScope.kt [hunk 2]
```diff
@@ -33,4 +33,4 @@ class VertxDispatcher(private val vertxContext: Context) : CoroutineDispatcher()
             block.run()
```
Comment (cescoffier) [inline extensions/resteasy-reactive/quarkus-resteasy-reactive/kotlin/src/main/kotlin/org/jboss/resteasy/reactive/server/runtime/kotlin/ApplicationCoroutineScope.kt:33]: shouldn't this be checking that we are on the same context as the one passed above?

File: extensions/resteasy-reactive/quarkus-resteasy-reactive/deployment/src/test/java/io/quarkus/resteasy/reactive/server/test/kotlin/BlockingCoroutinesTest.java (hunk 1 above)
Comment (geoand) [reply extensions/resteasy-reactive/quarkus-resteasy-reactive/deployment/src/test/java/io/quarkus/resteasy/reactive/server/test/kotlin/BlockingCoroutinesTest.java:39]: Even in the tests?

File: extensions/resteasy-reactive/quarkus-resteasy-reactive/kotlin/src/main/kotlin/org/jboss/resteasy/reactive/server/runtime/kotlin/ApplicationCoroutineScope.kt (hunk 2 above)
Comment (geoand) [reply extensions/resteasy-reactive/quarkus-resteasy-reactive/kotlin/src/main/kotlin/org/jboss/resteasy/reactive/server/runtime/kotlin/ApplicationCoroutineScope.kt:33]: Good point. I didn't really change this, this what is what we already have

File: extensions/resteasy-reactive/quarkus-resteasy-reactive/deployment/src/test/java/io/quarkus/resteasy/reactive/server/test/kotlin/BlockingCoroutinesTest.java (hunk 1 above)
Comment (cescoffier) [reply extensions/resteasy-reactive/quarkus-resteasy-reactive/deployment/src/test/java/io/quarkus/resteasy/reactive/server/test/kotlin/BlockingCoroutinesTest.java:39]: Yes, even in the tests. That will complexity the build pipeline.

Comment (geoand) [reply extensions/resteasy-reactive/quarkus-resteasy-reactive/deployment/src/test/java/io/quarkus/resteasy/reactive/server/test/kotlin/BlockingCoroutinesTest.java:39]: Okay, I will then move this test to the integration tests module

Comment (geoand) [reply extensions/resteasy-reactive/quarkus-resteasy-reactive/deployment/src/test/java/io/quarkus/resteasy/reactive/server/test/kotlin/BlockingCoroutinesTest.java]: I added a new module for these tests

File: extensions/resteasy-reactive/quarkus-resteasy-reactive/kotlin/src/main/kotlin/org/jboss/resteasy/reactive/server/runtime/kotlin/ApplicationCoroutineScope.kt (hunk 2 above)
Comment (mschorsch) [reply extensions/resteasy-reactive/quarkus-resteasy-reactive/kotlin/src/main/kotlin/org/jboss/resteasy/reactive/server/runtime/kotlin/ApplicationCoroutineScope.kt:33]: @cescoffier I am not sure if I understand you correctly, but are you aware of this [discussion](https://github.com/quarkusio/quarkus/pull/16931#discussion_r624789718)?

Comment (cescoffier) [reply extensions/resteasy-reactive/quarkus-resteasy-reactive/kotlin/src/main/kotlin/org/jboss/resteasy/reactive/server/runtime/kotlin/ApplicationCoroutineScope.kt:33]: Yes, I've seen it. I'm not totally convinced by the "do not call this method immediately". In general, it's a good optimization avoiding going back to the waiting queue. I can understand why they want to enforce this, but in Vert.x typically, we often decide to call immediately if we are already on the right context for performance reasons.

**Review by stuartwdouglas — APPROVED** [review]

**Pull request merged**